### ISSUE TITLE
Modules can shadow builtin types like str

### ISSUE THREAD

**Issue by kumbayo** [intellij-rust/intellij-rust#2228]
At places where a type is expected, we can erroneously also resolve to non type items (e.g. mod).

This was originally reported at https://users.rust-lang.org/t/rust-beginner-notes-questions/14928/6
as `use nom::*;` breaking the str primitive type.

From what i can see Rust actually knows if the resolved target should be a type, function, trait, module, .... and ignores unexpected targets with the same name if a expected one is found.
When only unexpected targets are found an error is output. 

See
https://github.com/rust-lang/rust/blob/48a7ea9c4095e3218df39832dfc51a456534ecc9/src/librustc_resolve/lib.rs#L401
https://github.com/rust-lang/rust/blob/48a7ea9c4095e3218df39832dfc51a456534ecc9/src/librustc_resolve/lib.rs#L2631

Test code (which compiles successfully):
```rust
fn nom_not_shadowed_1() {
    use nom::*;
    let s: &str = unreachable!(); //IntelliJ complains: "Module 'str' is private". We should not resolve to it since it is a mod and not a type...
    s.to_string();
}

mod contains_public_mod {
    pub mod str {}
}

mod contains_private_mod {
    mod str {}
}

fn not_shadowed1() {
    mod str {}
    let s: &str/*should not resolve to the mod*/ = unreachable!();
    s.to_string();
}

fn not_shadowed2() {
    use self::contains_public_mod::str;
    let s: &str/*should not resolve to the mod*/ = unreachable!();
    s.to_string();
}

fn not_shadowed3() {
    use self::contains_public_mod::*;
    let s: &str/*should not resolve to the mod*/ = unreachable!();
    s.to_string();
}

fn not_shadowed4() {
    use self::contains_private_mod::*;
    //Complains because it is private but should probably just not import it...
    let s: &str/*should not resolve to the mod*/ = unreachable!();
    s.to_string();
}

mod contains_struct {
    pub struct str {}
    impl str {
        pub fn foo(&self) {}
    }
}

fn shadowed1() {
    use self::contains_struct::str;
    let s: &str/*should resolve to the struct*/ = unreachable!();
    s.foo();
}
```
